src/index.js

```diff
@@ -36,9 +36,7 @@ export default function Composer({
             // Remove the current component and continue.
             components.slice(1),
             // results.concat([mapped]) ensures [...results, mapped] instead of [...results, ...mapped]
-            results.concat(
-              mapResult ? [mapResult.apply(null, arguments)] : arguments[0]
-            )
+            results.concat([mapResult.apply(null, arguments)])
```

Comment:
🤔  I will say that I can see taking this a step further and deprecating/dropping the concept of `props.mapResult` altogether, implementing this as core behavior so that:

```diff
-            results.concat([mapResult.apply(null, arguments)])
+            results.concat([arguments.length === 1 ? arguments[0] : Array.from(arguments)])
```

So this would no longer configurable and `react-composer` just respects the produced signature of each render prop component.

Typical usage (all single-arg producers): 

```jsx
<Composer {/* ... */}>
  {([one, two, three]) => /* ... */}
</Composer>
```

Edge case (multi-arg producer(s) involved):

```jsx
<Composer {/* ... */}>
  {([one, [twoA twoB], three]) => /* ... */}
</Composer>
```

I feel like that could be reasonable behavior considering that if you are involving a multi-arg producer in your composition you will surely be aware of it and would not desire the additional arguments to be discarded... keeping in mind this is the exception and optimizing for (the 90%+?) single-arg producers.